index.js

```diff
@@ -2,105 +2,77 @@
 'use strict';
 const Funnel = require('broccoli-funnel');
 const getDebugMacros = require('./src/debug-macros').debugMacros;
-const semver = require('semver');
-const pkg = require('./package.json');
+
+const VersionChecker = require('ember-cli-version-checker');
 
 /*
  shouldIncludeChildAddon runs during
  addon init so does not have access to
  the host application instance.
 */
 let requiredDataPackages = ['@ember-data/model', '@ember-data/store', 'ember-inflector'];
-let allDataPackages = [
-  '@ember-data/adapter',
-  '@ember-data/debug',
-  '@ember-data/model',
-  '@ember-data/record-data',
-  '@ember-data/serializer',
-  '@ember-data/store',
-];
-let minVersionForUsingOwnPackages = '3.14.0';
-
-function hasDataPackage(addon) {
-  let addons = addon.project.addonPackages;
-  return addons['ember-data'] !== undefined;
-}
-function hasOwnDataPackages(addon) {
-  let addons = addon.project.addonPackages;
 
-  for (let i = 0; i < allDataPackages.length; i++) {
-    if (addons[allDataPackages[i]] !== undefined) {
-      return true;
+function assertValidEmberData(addon) {
```

Comment:
We should likely generalize this, and move this or its constituent parts to ember-cli-version-checker and thoroughly test it.